include/swift/AST/FineGrainedDependencies.h

```diff
@@ -684,15 +689,15 @@ class SourceFileDepGraphNode : public DepGraphNode {
   /// True iff a Decl exists for this node.
   /// If a provides and a depends in the existing system both have the same key,
   /// only one SourceFileDepGraphNode is emitted.
-  bool isProvides;
+  bool isProvides = false;
```

Comment:
Without this the no-argument constructor produces an object where 'isProvides' has an undefined value. I'm not sure that warrants a comment.

Reply:
When I saw the "false" I figured that the `isProvides` field would only be set if it were `true`, and you were using the `false` because if never set, that meant it would be `false`. If such is the case, the code would be clearer to me with a comment.

If, on the other hand, it is the case that this field will be set to either truth value, and the initialization to `false` has no effect, then, the way I read code, I would find this initialization confusing without a comment. The initializer sets the sequence number to an illegal value so that initialized-but-never-set nodes can be  detected.